eng/common/scripts/Invoke-DevOpsAPI.ps1

```diff
@@ -2,6 +2,24 @@
 
 $DevOpsAPIBaseURI = "https://dev.azure.com/{0}/{1}/_apis/{2}/{3}?{4}api-version=6.0"
 
+function Get-Base64EncodedToken([string]$AuthToken)
+{
+  $unencodedAuthToken = "nobody:$AuthToken"
+  $unencodedAuthTokenBytes = [System.Text.Encoding]::UTF8.GetBytes($unencodedAuthToken)
+  $encodedAuthToken = [System.Convert]::ToBase64String($unencodedAuthTokenBytes)
+
+  if ($isDevOpsRun) {
+    # We are doing this here so that there is zero chance that this token is emitted in Azure Pipelines
+    # build logs. Azure Pipelines will see this text and register the secret as a value it should *** out
+    # before being transmitted to the server (and shown in logs). It means if the value is accidentally
+    # leaked anywhere else that it won't be visible. The downside is that when the script is executed
+    # on a local development box, it will be visible.
```

Comment:
> on a local development box, it will be visible.

Maybe we could add some additional check to see if we run on local devbox, and if so, do not apply this protection? Not sure if worth the complexity, though.

Reply:
That is what the isDevOpsRun check is about. That shouldn't be set on local dev box.

Reply:
> That is what the isDevOpsRun check is about. That shouldn't be set on local dev box.

OK then in such case this comment:

> The downside is that when the script is executed on a local development box, it will be visible.

Is a bit confusing to me. Because I understand you are saying that it is extremely unlikely that `$isDevOpsRun` will be set in local run. Then why mention this risk of outputting things locally? OTOH if it is real risk, then we could have something like `if ($isDevOpsRun && <running_locally>) then { throw }`

Reply:
Fair let me clean up that comment. I honestly didn't give it much thought as I just copied it from the other place but I should clean it up.